presentation/src/main/java/me/androidbox/presentation/worker/UploadEventWorker.kt

```diff
@@ -0,0 +1,105 @@
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.presentation.util.CreatePhotoMultipart
+import okhttp3.MultipartBody
+import java.util.*
+import kotlin.random.Random
+
+/*
+content://com.android.providers.media.documents/document/image%3A38325
+content://com.android.providers.media.documents/document/image%3A38324
+content://com.google.android.apps.photos.contentprovider/0/1/mediakey%3A%2Flocal%253A44f502c3-7f36-4f9b-887c-5d8f2f45f3cc/ORIGINAL/NONE/image%2Fjpeg/430889281
+content://com.google.android.apps.photos.contentprovider/-1/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F37452/ORIGINAL/NONE/image%2Fjpeg/640889962
+content://com.google.android.apps.photos.contentprovider/0/1/mediakey%3A%2Flocal%253A1e78ee30-5b74-4a1e-862b-a13525f26cd4/ORIGINAL/NONE/image%2Fjpeg/63330985
+content://com.google.android.apps.photos.contentprovider/0/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F38330/ORIGINAL/NONE/image%2Fjpeg/1631907737
+content://com.android.providers.media.documents/document/image%3A38324
+content://com.android.providers.media.documents/document/image%3A38337
+* */
+
+@HiltWorker
+class UploadEventWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val eventService: EventService,
+    private val eventConverter: EventConverter,
+    private val createPhotoMultipart: CreatePhotoMultipart
+) : CoroutineWorker(context, workerParameters)  {
+
+    override suspend fun doWork(): Result {
+        startForegroundService()
+
+        /** TODO Remove this mock data - this is just for testing purposes */
+        val eventRequest = EventRequestDto(
+            id = UUID.randomUUID().toString(),
+            title = "Title",
+            description = "description",
+            from = 3L,
+            to = 6L,
+            remindAt = 4L,
+            attendeeIds = listOf(
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString(),
+                UUID.randomUUID().toString()
+            )
+        )
+
+        /* Serialize the event request to json */
+        val eventRequestJson = eventConverter.toJson(eventRequest)
+
+        /* These are just a random set of photos that I got from selecting photos from the device.
+         * I just copied them here to try and send them as a multi-part request just to test if I can send them */
+        val listOfPhotos = listOf(
+            "content://com.google.android.apps.photos.contentprovider/0/1/mediakey%3A%2Flocal%253A44f502c3-7f36-4f9b-887c-5d8f2f45f3cc/ORIGINAL/NONE/image%2Fjpeg/430889281",
+            "content://com.android.providers.media.documents/document/image%3A38324",
+            "content://com.google.android.apps.photos.contentprovider/-1/1/content%3A%2F%2Fmedia%2Fexternal%2Fimages%2Fmedia%2F37452/ORIGINAL/NONE/image%2Fjpeg/640889962")
```

Comment:
AFAIK, these may differ each time you retrieve them. Not tested, but have heard people say that. I'd test this with a Uri you really picked on the device (this could explain your `FileNotFoundException`).